dist_run.py

```diff
@@ -294,7 +309,7 @@ def main(args):
     config = TransformerArgs.from_params(model_config.transformer_args["text"])
     logger.info(f"Transformer Config: {config}")
 
-    tokenizer = _build_chat_tokenizer(model_name)
+    tokenizer, tokenizer_type = _build_chat_tokenizer(model_name)
```

Comment:
nit: can you make `tokenizer_type` a global so that we can also use it in functions like `_decode_in_flight` ?
`_build_chat_tokenizer` will set this global, and will not need to return it.

Reply:
I've updated to make tokenizer_type a global (and renamed to _tokenizer_type as a type hint to it's global state).